test-support/src/main/java/org/springframework/ldap/test/unboundid/EmbeddedLdapServer.java

```diff
@@ -102,4 +107,88 @@ public void shutdown() {
 		this.directoryServer.shutDown(true);
 	}
 
+	/**
+	 * Helper class for embedded Unboundid ldap server.
+	 *
+	 * @author Emanuel Trandafir
+	 * @since 3.3
+	 */
+	public static final class Builder {
+
+		private int port = 0;
+
+		private Consumer<InMemoryDirectoryServerConfig> configurationCustomizer = (__) -> {
+		};
+
+		private String partitionName = "jayway";
```

Comment:
I think this partition name is too specific. Instead, I think you can by default make an educated guess that the partition name is the left-most element of the partition suffix. In the constructor, perhaps consider using `LdapName` to parse the suffix and extract that element.